src/main/java/bio/terra/app/logging/LoggerInterceptor.java

```diff
@@ -14,40 +16,75 @@
 
 @Component
 public class LoggerInterceptor implements HandlerInterceptor {
-  private static Logger logger = LoggerFactory.getLogger(LoggerInterceptor.class);
+  private static final Logger logger = LoggerFactory.getLogger(LoggerInterceptor.class);
+  // Constants for requests coming in that aren't authenticated
+  private static final String UNAUTHED_USER_ID = "N/A";
+  private static final String UNAUTHED_EMAIL = "N/A";
+  private static final String UNAUTHED_INSTITUTE = "N/A";
+  // Don't log requests for URLs that end with any of   the following paths
+  private static final Set<String> LOG_EXCLUDE_LIST = Set.of("/status");
+
+  private static final String REQUEST_START_ATTRIBUTE = "x-request-start";
+  private static final Long NOT_FOUND_DURATION = -1L;
+
   private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
 
   @Autowired
   public LoggerInterceptor(AuthenticatedUserRequestFactory authenticatedUserRequestFactory) {
     this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
   }
 
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+      throws Exception {
+    request.setAttribute(REQUEST_START_ATTRIBUTE, System.currentTimeMillis());
+    return true;
+  }
+
   @Override
   public void afterCompletion(
       HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex)
       throws Exception {
-    AuthenticatedUserRequest userReq = authenticatedUserRequestFactory.from(request);
-    String userId = userReq.getSubjectId();
-    String userEmail = userReq.getEmail();
-    String institute = userEmail != null ? userEmail.substring(userEmail.indexOf("@") + 1) : null;
+
+    String userId;
+    String userEmail;
+    String institute;
+    try {
+      AuthenticatedUserRequest userReq = authenticatedUserRequestFactory.from(request);
+      userId = userReq.getSubjectId();
+      userEmail = userReq.getEmail();
+      institute = userEmail != null ? userEmail.substring(userEmail.indexOf("@") + 1) : null;
+    } catch (UnauthorizedException e) {
+      userId = UNAUTHED_USER_ID;
+      userEmail = UNAUTHED_EMAIL;
+      institute = UNAUTHED_INSTITUTE;
+    }
+
     String url = request.getRequestURL().toString();
     String method = request.getMethod();
     Map<String, String[]> paramMap = request.getParameterMap();
     Gson gson = new Gson();
     String paramString = gson.toJson(paramMap);
     String responseStatus = Integer.toString(response.getStatus());
-
+    long requestDuration;
+    Long requestStartTime = (Long) request.getAttribute(REQUEST_START_ATTRIBUTE);
```

Comment:
Under what conditions is this value missing? Do we expect this to happen often?

Reply:
Under the condition where I'm paranoid...